src/adapter/src/optimize/peek.rs

```diff
@@ -407,30 +408,19 @@ impl<'s> Optimize<GlobalMirPlan<ResolvedGlobal<'s>>> for Optimizer {
             Some(&self.finishing),
             self.config.persist_fast_path_limit,
         )? {
-            Some(plan) => {
-                // An ugly way to prevent panics when explaining the physical
-                // plan of a fast-path query.
-                //
-                // TODO: get rid of this.
-                if self.config.mode == OptimizeMode::Explain {
-                    // Finalize the dataflow. This includes:
-                    // - MIR ⇒ LIR lowering
-                    // - LIR ⇒ LIR transforms
-                    let _ = Plan::<Timestamp>::finalize_dataflow(
-                        df_desc,
-                        self.config.enable_consolidate_after_union_negate,
-                        self.config.enable_specialized_arrangements,
-                        self.config.enable_reduce_mfp_fusion,
-                    )
-                    .map_err(OptimizerError::Internal)?;
-                }
+            Some(plan) if self.config.enable_fast_path => {
+                // Trace the pipeline input under `optimize/raw`.
+                debug_span!(target: "optimizer", "fast_path").in_scope(|| {
```

Comment:
This and the following `trace_plan` calls correspond to code removed from `inner/peek.rs`.